src/components/DataDimension/DataDimension.js

```diff
@@ -77,8 +113,47 @@ const DataDimension = ({
                 infoBoxMessage={infoBoxMessage}
                 dataTest={'data-dimension'}
                 dataTypes={dataTypes}
+                itemsRef={itemsRef}
+                supportsEDI={supportsEDI}
                 onEDISave={onCalculationSave}
+                isOptionViewMode={Boolean(currentDataItem)}
+                currentCalculation={currentCalculation}
+                setCurrentCalculation={setCurrentCalculation}
+                infoDataItem={infoDataItem}
+                setInfoDataItem={setInfoDataItem}
+                onEditClick={onEditClick}
             />
+            {currentDataItem && (
+                <ItemOptionsSelector
+                    {...currentDataItem}
+                    selectedItems={selectedDimensions.map((item) => ({
```

Comment:
@edoardo This is recreated every time. Could it be memo'ed to avoid subsequent re-renders?

Reply:
Yes, extracted the map into a variable and used in both components.